wdl-format/src/v1.rs

```diff
@@ -1,66 +1,328 @@
 use crate::element::FormatElement;
-use crate::exactly_one;
 
 /// Formats an [`Ast`](wdl_ast::Ast).
 pub fn format_ast(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
-    let mut children = element.children_by_kind();
+    let mut children = element.children().expect("AST children");
 
-    if let Some(mut versions) = children.remove(&SyntaxKind::VersionStatementNode) {
-        let version = exactly_one!(versions, "version statements");
+    let version_statement = children.next().expect("version statement");
+    assert!(version_statement.element().kind() == SyntaxKind::VersionStatementNode);
+    (&version_statement).write(stream);
 
-        // TODO(clay): improve this by removing the reference.
-        (&version).write(stream);
-    }
+    stream.blank_line();
 
-    stream.section_spacer();
+    let mut imports = Vec::new();
+    let mut remainder = Vec::new();
 
-    if let Some(tasks) = children.remove(&SyntaxKind::TaskDefinitionNode) {
-        for task in tasks {
-            (&task).write(stream);
-            stream.section_spacer();
+    for child in children {
+        match child.element().kind() {
+            SyntaxKind::ImportStatementNode => imports.push(child),
+            _ => remainder.push(child),
         }
     }
 
-    if let Some(workflows) = children.remove(&SyntaxKind::WorkflowDefinitionNode) {
-        for workflow in workflows {
-            (&workflow).write(stream);
-            stream.section_spacer();
-        }
+    imports.sort_by(|a, b| {
+        let a = a
+            .element()
+            .as_node()
+            .expect("import statement node")
+            .as_import_statement()
+            .expect("import statement");
+        let b = b
+            .element()
+            .as_node()
+            .expect("import statement node")
+            .as_import_statement()
+            .expect("import statement");
+        let a_uri = a.uri().text().expect("import uri");
+        let b_uri = b.uri().text().expect("import uri");
+        a_uri.as_str().cmp(b_uri.as_str())
+    });
+
+    for import in imports {
+        (&import).write(stream);
     }
 
-    if !children.is_empty() {
-        todo!("unhandled children for AST: {:#?}", children.keys());
+    stream.blank_line();
+
+    for child in remainder {
+        (&child).write(stream);
+        stream.blank_line();
     }
 }
 
 /// Formats a [`VersionStatement`](wdl_ast::VersionStatement).
 pub fn format_version_statement(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
-    let mut children = element.children_by_kind();
+    for child in element.children().expect("version statement children") {
+        (&child).write(stream);
+        stream.end_word();
+    }
+    stream.end_line();
+}
+
+/// Formats a [`LiteralString`](wdl_ast::v1::LiteralString).
+pub fn format_literal_string(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
+    for child in element.children().expect("literal string children") {
```

Comment:
I'm just placing this comment here because it's where I thought of it (not because it has to be here exactly): we may want to slightly change the way that the `FormatElement` and `Writable` trait works.

In general, what we want to ensure that, when an element is formatted, _all_ of the children of that element *must* be visited (even if they are explicitly not formatted). This suggests to me that `FormatElement` should keep track of which children are visited in some way (either by explicitly marking the children or some other way that makes sense) and then check that the children have been exhaustively formatted before the `FormatElement` is `drop`ed (panic if that isn't the case).

Reply:
I'm going to leave that to you to think through and decide on.